common/src/utils/camera.rs

```diff
@@ -0,0 +1,100 @@
+use nalgebra::Matrix4;
+
+use crate::{
+    desktop::{InputEvent, InputEvents, WindowEvent},
+    vr::{VrFov, VrUpdate},
+};
+
+/// Perspective camera matrix utility
+/// In desktop mode, makes sure the matrix has the correct aspect ratio
+/// In VR mode, makes sure the perspective matrices match the XrFov specification
+pub struct Perspective {
+    screen_size: (u32, u32),
+    pub near_plane: f32,
+    pub far_plane: f32,
+    pub fov: f32,
+    proj: [Matrix4<f32>; 2],
+}
+
+impl Default for Perspective {
+    fn default() -> Self {
+        Self {
+            screen_size: (1920, 1080),
+            near_plane: 0.01,
+            far_plane: 1000.,
+            fov: 45_f32.to_radians(),
+            proj: [Matrix4::identity(); 2],
+        }
+    }
+}
+
+impl Perspective {
+    /// Initialize the default perspective
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Returns the left and right perpective matrices, respectively.
+    /// In desktop mode these matrices will be equal
+    pub fn matrices(&self) -> [Matrix4<f32>; 2] {
+        self.proj
+    }
+
+    /// Returns and appropriate perspective matrix matching the size of the window
+    pub fn handle_input_events(&mut self, input: &InputEvents) {
+        // Handle input events for desktop mode
+        let InputEvents(events) = input;
+        for event in events {
+            if let InputEvent::Window(WindowEvent::Resized { width, height }) = event {
+                self.screen_size = (*width, *height);
+            }
+        }
+
+        // Get projection matrix
+        let proj = Matrix4::new_perspective(
+            self.screen_size.0 as f32 / self.screen_size.1 as f32,
+            self.fov,
+            self.near_plane,
+            self.far_plane,
+        );
+
+        self.proj = [proj; 2];
+    }
+
+    pub fn handle_vr_update(&mut self, update: &VrUpdate) {
```

Comment:
This function is slightly confusing, maybe add a comment or perhaps rename the VrUpdate struct. From an initial glance it seems to take whatever the `VrUpdate` items are and perform the fov projection on that but it's unclear what information is stored in `VrUpdate` based off the name.

Reply:
Just gonna leave this the same as it is for now, this whole `VrUpdate` thing needs an overhaul tbh

Reply:
Sounds good!